node/src/components/chain_synchronizer/operations.rs

```diff
@@ -1553,6 +1559,64 @@ where
                     );
                     return Err(error);
                 }
+
+                let query_request = QueryRequest::new(
+                    trusted_state_root_hash.clone(),
+                    Key::DeployApprovalsRootHash { block_height },
+                    vec![],
+                );
+
+                let query_result = ctx
+                    .effect_builder
+                    .query_global_state(query_request)
+                    .await
+                    .map_err(Error::ExecutionEngine)?;
+                match query_result {
+                    QueryResult::RootNotFound => {
+                        return Err(Error::TrustedHeaderStateRootNotFound);
+                    } // Error
+                    QueryResult::ValueNotFound(_err) => {}
+                    QueryResult::CircularReference(_err) => todo!(), // TODO Ask
+                    QueryResult::DepthLimit { depth: _ } => todo!(), // TODO Ask
+                    QueryResult::Success { value, proofs: _ } => {
+                        let deploy_approvals_root_hash: Digest = match *value {
+                            StoredValue::CLValue(cl_value) => cl_value
+                                .into_t()
+                                .map_err(|cl_value_error| Error::CLValueError { cl_value_error })?,
+                            stored_value => {
+                                return Err(Error::WrongTypeUnderDeployApprovalsRootHash {
+                                    stored_value,
+                                });
+                            }
+                        };
+
+                        let mut approval_hashes = vec![];
+                        for digest in fetched_block
+                            .block
+                            .deploy_hashes()
+                            .iter()
+                            .map(|deploy_hash| deploy_hash.inner().clone())
+                            .chain(
+                                fetched_block
+                                    .block
+                                    .transfer_hashes()
+                                    .iter()
+                                    .map(|transfer_hash| transfer_hash.inner().clone()),
+                            )
+                        {
+                            approval_hashes.push(digest);
+                        }
```

Comment:
I may be misunderstanding, but aren't you collecting the deploy and transfer hashes here, instead of approvals (or approval hashes)?